### ISSUE TITLE
[Question] Did someone try to post-produce audio, equalizer etc.

### ISSUE THREAD

**Issue by zevarito** [ging/licode#312]
I found that the recorded audio quality is pretty bad, but I wonder if it could improve with some sort of post-processing.

**Comment by zevarito** [comment]
@lodoyun Thanks, we already have done that but it still sounds like inside a pipe and with high levels of trebles.

* Should be `pcmu` commented at the same time `opus` become enabled ?
* How Licode decides which codec to use?

I am digging into `opus` documentation to see if there are parameters we can tune to get even better results.

Here is the output of `ffmpeg -i` of a recored MKV with `opus` enabled.

```
Stream #0:0: Video: vp8, yuv420p, 640x480, SAR 1:1 DAR 4:3, 1k tbr, 1k tbn, 1k tbc
Stream #0:1: Audio: opus, 48000 Hz, stereo, fltp
```

**Comment by lodoyun** [comment]
Is the quality in the live session is better than the recording?

Licode does not actually decide which codec to use, browsers decide it. We only leave one option open (PCMU or OPUS) to make sure all the session uses the same codec, however firefox and Chrome should pick OPUS, so for a quick test it's ok to leave both codecs there.

**Comment by zevarito** [comment]
Yes, the quality in live session is considerable better than the recorded one.

I have the following questions:

* It seems like audio is recored at something like 8bit 8000Hz and then resampled into 48000Hz, could be?.
* Is the MCU responsible for recording? Does it pick an audio codec as-well?
* It is possible to let him pick PCMU with 48000 rate and 2 channels, instead of 8000 and 1?

**Comment by lodoyun** [comment]
Audio is recorded exactly how it is received in Erizo from the browser(no transcoding or resampling), that is 8000 Hz mono for PCMU and 48000 / 2 channels for OPUS.
There is no way to change the Sample frequency of PCMU
If you are using OPUS you already have 48000/2
You are post-processing the file right? If you are transcoding to webm/mp4 you are transcoding the audio to vorbis/aac/mp3, check the parameters just in case.

**Comment by zevarito** [comment]
Yes, I am post processing the file using Amazon ET, but I don't see differences when I used FFMPEG.

I am looking at the RFC's, seems to be possible to alter Hz, but more important to define Thresholds for Opus Hz. I think is worth to experiment with other codec settings to see if quality improves.

If I end with something I let you know.

Also I was thinking

**Comment by joshuairl** [comment]
I'm having this problem too.
The quality of sound in the recording does seem to be worse in the recording than when talking to someone live but I guess I could be just remembering correctly.

Are you saying it should be identical as the live stream or...?

**Comment by joshuairl** [comment]
Silly question but I'm assuming after changing the cpp we need to recompile? :D haha
